modules/salt/pages/salt-rate-limiting.adoc

```diff
@@ -0,0 +1,60 @@
+[[salt.rate.limiting]]
+= Rate Limiting
+
+Salt is able to run commands in parallel on a large number of minions.
+This can potentially create large amounts of load on your infrastructure.
+You can use these rate-limiting parameters to control the load in your environment.
+
+These parameters are all configured in the [filename]``/etc/rhn/rhn.conf`` configuration file.
+
+[NOTE]
+====
+Salt commands that are executed from the command line are not subject to these parameters.
+====
+
+
+
+== Batch Size
+
+This parameter sets the maximum number of clients that can execute a single action at the same time.
+
+Choosing a lower number will reduce system load and parallelism, but might reduce overall performance for processing actions.
+
+Adjust the [systemitem]``java.salt_batch_size`` parameter.
+Defaults to 100.
+
+
+
+== Batch Delay
+
+This parameter sets the amount of time, in seconds, to wait after an action is processed before beginning to process the action on the next minion.
+To begin, actions are processed on the number of minions determined in the batch size parameter.
+As each minion completes, the action begins on a new minion after the specified delay period.
+
+Increasing the delay increases the chance that multiple minions will have completed before the next action is issued, resulting in fewer overall commands, and reducing load.
+
+Adjust the [systemitem]``java.salt_batch_delay`` parameter.
+Defaults to 1.0 seconds.
+
+
+
+== Ping Timeout
+
+During Salt batch calls, Salt clients will perform a presence ping.
+This parameter changes the timeout setting for the presence ping, in seconds.
+
+Adjust the [systemitem]``java.salt_presence_ping_timeout`` parameter.
+Defaults to 4 seconds.
+
+
+
+== Job Timeout
+
+During Salt batch calls, a job runs to handles the presence ping from the clients.
+This parameter changes the timeout setting for gathering the presence ping, in seconds.
+
+If a minion does not reply to a ping within the allocated time, it will be marked as [systemitem]``not available``, and will be excluded from the command.
```

Comment:
This note is basically a conclusion of both the `salt_presence_ping_timeout` and the `salt_presence_ping_gather_job_timeout`.

What happens is well described here: https://www.suse.com/documentation/suse-manager-3/book_suma_best_practices_3/data/bp_chap_salt_minion_scaleability_timeouts.html

I think it is worth to put that detailed description here as well - otherwise it's hard to understand why there are two distinct timeout settings and what their mean is.

Note that the logic there about `timeout`/`gather_job_timeout` (total default: 240 seconds) is the same applied for `salt_presence_ping_timeout`/`salt_presence_ping_gather_job_timeout` (total default: 5 seconds) - only that the latter couple only affects presence pings. We want to be able to set those differently to more quickly detect minions that are down while also allowing for much wider margins for non-ping commands.

Reply:
This content exists in the client cfg guide now, but I think it's better suited to the Salt Guide. I'll move it in.